demos/common/monitors/presenter.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2019 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <map>
+#include <ostream>
+#include <set>
+
+#include <opencv2/imgproc.hpp>
+
+#include "cpu_monitor.h"
+#include "memory_monitor.h"
+
+enum class MonitorType{CpuAverage, DistributionCpu, Memory};
+
+class Presenter {
+public:
+    explicit Presenter(std::set<MonitorType> enabledMonitors = {},
+        int yPos = 20,
+        cv::Size graphSize = {150, 60},
+        std::size_t historySize = 20);
+    explicit Presenter(const std::string& keys,
+        int yPos = 20,
+        cv::Size graphSize = {150, 60},
+        std::size_t historySize = 20);
+    operator std::string() const;
+    void addRemoveMonitor(MonitorType monitor);
+    void addRemoveMonitor(int key); // handles c, d, m, h keys
+    void drawGraphs(cv::Mat& frame);
+
+    friend std::ostream& operator<<(std::ostream& os, const Presenter& presenter);
```

Comment:
I don't think using the output operator is appropriate here. You're not printing the presenter, you're printing the metrics.

Reply:
updated